src/app/components/Video/index.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { videoPropTypes, videoDefaultPropTypes } from '../../proptypes';
+import filterForBlockType from '../../../utils/blockHelpers';
+
+const Video = ({ model }) => {
+  const subBlocks = model.blocks;
+
+  const rawVideo = filterForBlockType(subBlocks, 'rawVideo');
+  if (!rawVideo) {
+    return null;
+  }
+
+  const { locator: videoLocator, duration, versionID, kind } = rawVideo.model;
+
+  const imageBlock = filterForBlockType(subBlocks, 'image');
+  const rawImage = filterForBlockType(imageBlock.model.blocks, 'rawImage');
+
+  if (!rawImage) {
+    return null;
```

Comment:
This seems untested

Reply:
I know this might be a streach, but could we move testing this to a separate PR, I just found that there is a bug in the test block constructor that kills the test if no image block is specified.